libkbfs/folder_branch_ops.go

```diff
@@ -1911,7 +1987,7 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		} else {
 			bid = fbo.bid
 		}
-		err := mdops.PutUnmerged(ctx, md, bid)
+		mdID, err = fbo.doPutWithCancellationCheck(ctx, lState, md, bid)
```

Comment:
It's the same context that is already "critical aware", which had a timeout on cancellation (if any) equal to `GracePeriod()`. Thinking about this together with the comment below on adding `ExitCritical`, shall we maybe do `ExitCritical` at the end of `finalizeMDWriteLocked`?